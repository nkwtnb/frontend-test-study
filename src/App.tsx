import axios from 'axios';
import { useState } from 'react';
import './App.css';
import Counter from './components/Counter';
import Spacer from './components/Spacer';


function App() {

  const [posts, setPosts] = useState([]);

  const handleClick = async () => {
    const resp = (await axios.get("https://jsonplaceholder.typicode.com/posts")).data;
    setPosts(resp);
  }
  
  return (
    <div className="App">
      <header className="App-header">
        <div >
          <div>
            カウンター：<Counter />
          </div>
        </div>
        <Spacer />
        <div>
          <div>API：</div>
          <button data-testid="get-api" onClick={handleClick}>データ取得</button>
          <div data-testid="posts" id="posts">
            {
              posts.map((post: any, index: number) => {
                return (
                  <div className='post' key={index}>{post.id}</div>
                )
              })
            }
          </div>
        </div>
      </header>
    </div>
  );
}

export default App;
