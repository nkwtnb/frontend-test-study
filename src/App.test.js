import { render, screen } from '@testing-library/react';
import App from './App';

describe("test block", () => {
  test("test counter", () => {
    render(<App />);
    screen.getByTestId("add").click()
    expect(screen.getByTestId("counter").innerHTML).toBe("1");
    screen.getByTestId("minus").click()
    expect(screen.getByTestId("counter").innerHTML).toBe("0");
  });
});

describe("snapshot test block", () => {
  test('snapshot test', () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
