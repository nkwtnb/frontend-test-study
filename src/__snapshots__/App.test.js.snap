// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test block snapshot test 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div>
        <span
          data-testid="counter"
        >
          0
        </span>
        <button
          data-testid="add"
        >
          ＋
        </button>
        <button
          data-testid="minus"
        >
          ー
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;

exports[`snapshot test snapshot test 1 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div>
        <span
          data-testid="counter"
        >
          0
        </span>
        <button
          data-testid="add"
        >
          ＋
        </button>
        <button
          data-testid="minus"
        >
          ー
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
